src/coffee/import.coffee

```diff
@@ -238,13 +239,37 @@ class Import
         else
           existingProduct.variants[variantInfo.index] = variant
         productsToUpdate[productIndex] =
-          product: existingProduct
+          product: @mergeProductLevelInfo existingProduct, _.deepClone entry.product
           header: entry.header
           rowIndex: entry.rowIndex
       else
         console.warn "Ignoring variant as no match by SKU found for: ", variant
     _.map productsToUpdate
 
+  mergeCategories: (target, source) ->
+    # function will try to find category by its ID in a given list
+    findById = (list, id) ->
+      _.find(list, (item) -> item.id == id)
+
+    target = target or []
+    source = source or []
+
+    # go through source list and push items
+    # which are not yet present in a target list
+    source.forEach (item) ->
```

Comment:
Can you please check how it behaves if the order of categories in the array do change (get a different index).